renderer/asciidoctor.go

```diff
@@ -143,7 +143,17 @@ func (adr *AsciidoctorRenderer) RenderAnchorID(id string) string {
 }
 
 func (adr *AsciidoctorRenderer) RenderFieldDoc(text string) string {
-	// escape the pipe character, which has special meaning for asciidoc as a way to format tables,
-	// so that including | in a comment does not result in wonky tables
-	return strings.ReplaceAll(text, "|", "\\|")
+	// Escape the pipe character, which has special meaning for asciidoc as a way to format tables,
+	// so that including | in a comment does not result in wonky tables.
+	out := strings.ReplaceAll(text, "|", "\\|")
+
+	// Trim any leading and trailing whitespace from each line.
+	lines := strings.Split(out, "\n")
+	for i := range lines {
+		lines[i] = strings.TrimSpace(lines[i])
+	}
+
+	// Replace newlines with hard line breaks so that newlines are rendered as expected.
+	// See: https://docs.asciidoctor.org/asciidoc/latest/blocks/hard-line-breaks
+	return strings.Join(lines, " +\n\n")
```

Comment:
Why an additional `\n`?
```suggestion
	return strings.Join(lines, " +\n")
```

Reply:
I made it this way, because the preprocessing of comments turns:

```go
	// Comment by guest. This can be a multi-line comment.
	// Which has multiple paragraphs.
        //
        // Just like this one.
```

into the following `FieldDoc` string:

```markdown
Comment by guest. This can be a multi-line comment. Which has multiple paragraphs.
Just like this one.
```

So we get a string with paragraphs separated with newlines. To preserve the intended formatting (so that paragraphs are actually separated) we have to replace newlines with doubled newlines to get:

```markdown
Comment by guest. This can be a multi-line comment. Which has multiple paragraphs.

Just like this one.
```